app/src/main/java/org/mozilla/focus/utils/StringExtensions.kt

```diff
@@ -0,0 +1,58 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.utils
+
+import android.net.Uri
+
+// Extension functions for the String class
+
+/**
+ * Beautify a URL by truncating it in a way that highlights important parts of the URL.
+ *
+ * Spec: https://github.com/mozilla-mobile/focus-android/issues/1231#issuecomment-326237077
```

Comment:
nit: the spec is contradictory: it mentions `parameter` and `value` should be special cased but then it always uses them in the examples (assuming those are get params) – should we add a comment to disambiguate?

Also, I see that it was changed in some places (e.g. nytimes only takes one parameter).

Also also, are get params always guaranteed to be in the same order (from my experience, I don't think they change without something changing on the server)? I wonder how this would affect our ability to display sites – should we sort them in some way?

Reply:
From what I have seen the order of get parameters is not defined. The existing APIs are just allowing to get parameters by name and not in a sorted way. That's why I'm falling back to splitting the string manually. From the spec I think we assume that whatever will be the first parameter might be the most important one (at least from the YouTube example) and so we will show at least the first one. I don't know of a better way to sort the parameters first. Sorting might lead to a different URL and not just a truncated one - which could be confusing.